ambari-server/src/main/resources/stacks/mpack_advisor.py

```diff
@@ -1689,31 +1689,28 @@ def getRecommendConfigurations(self, isSSO=False):
         mpackInstance = {}
         mpackInstance["name"] = mpackInstanceName
         mpackInstance["version"] = serviceInstances[0].getMpackVersion()
-        mpackInstance["service_instances"] = [serviceInstance.getName() for serviceInstance in serviceInstances]
-        mpackInstance["configurations"] = {}
-        mpackInstance["hosts"] = []
+        mpackInstance["service_instances"] = []
         mpackInstances.append(mpackInstance)
-        configurations = mpackInstance["configurations"]
         # there can be dependencies between service recommendations which require special ordering
         # for now, make sure custom services (that have service advisors) run after standard ones
         serviceAdvisors = []
         for serviceInstance in serviceInstances:
           serviceName = serviceInstance.getType()
```

Comment:
serviceName = serviceInstance.getName()

Also "type" is an optional parameter. So if "type" is not set in serviceInstance, then we need to set "type" = "name".